src/ext-test/java/org/opentripplanner/ext/fares/impl/GtfsFaresV2ServiceTest.java

```diff
@@ -146,23 +188,32 @@ void singleLeg() {
     Itinerary i1 = newItinerary(A, 0).walk(20, B).bus(ID, 0, 50, C).build();
 
     var result = service.getProducts(i1);
-    assertEquals(Set.of(dayPass, single), result.itineraryProducts());
+    assertEquals(
+      Stream
+        .of(single, dayPass, monthlyPass)
+        .flatMap(Collection::stream)
+        .collect(Collectors.toSet()),
+      result.itineraryProducts()
+    );
   }
 
   @Test
   void twoLegs() {
     Itinerary i1 = newItinerary(A, 0).walk(20, B).bus(ID, 0, 50, C).bus(ID, 55, 70, D).build();
 
     var result = service.getProducts(i1);
-    assertEquals(Set.of(dayPass), result.itineraryProducts());
+    assertEquals(
+      Stream.of(dayPass, monthlyPass).flatMap(Collection::stream).collect(Collectors.toSet()),
+      result.itineraryProducts()
+    );
   }
 
   @Test
   void networkId() {
     Itinerary i1 = newItinerary(A, 0).walk(20, B).faresV2Rail(ID, 0, 50, C, expressNetwork).build();
 
     var result = service.getProducts(i1);
-    assertEquals(Set.of(expressPass), result.itineraryProducts());
+    assertEquals(new HashSet<>(expressPass), result.itineraryProducts());
```

Comment:
Converting `Set` to `HashSet` is quite dubious. Can you explain what is going on here?